src/cache/diagnostic_edmf_precomputed_quantities.jl

```diff
@@ -310,6 +326,23 @@ function set_diagnostic_edmf_precomputed_quantities!(Y, p, t)
                 dt,
             )
 
+            # TODO:
+            # - we should also sum sgs contributions to energy and apply them
+            #   the same way as qt
+            # - code up the energy tendency and re-use it in here, depracated and precipitation
+
+            # Updraft q_tot sources from precipitation formation
+            # To be applied in updraft continuity, moisture and energy
+            # for updrafts and grid mean
+            @. S_q_totʲ_prev_level = -min(
```

Comment:
I think we guarantee that remove_precipitation is smaller than q_tot / dt by setting _τ_precip to dt, but maybe no harm to do it again here.